### PR TITLE
Making OutputPathCommandResolver work with MSBuild.

### PR DESCRIPTION
**Pull request by livarcocc**
@eerhardt @piotrpMSFT @jgoshi 

Fixes https://github.com/dotnet/cli/issues/4471

### CODE REVIEW

**Comment by piotrpMSFT** [comment]
This will conflict with the same change in my `3` change, but I'll make it work.

File: src/Microsoft.DotNet.Cli.Utils/CommandResolution/IProject.cs [hunk 1]
```diff
@@ -16,6 +16,8 @@ internal interface IProject
 
         string RuntimeConfigJsonPath { get; }
 
+        string OutputPath { get; }
```
Comment (eerhardt) [inline src/Microsoft.DotNet.Cli.Utils/CommandResolution/IProject.cs]: What do you think about calling this `FullOutputPath`?  It feels odd that this is named the exact same as an MSBuild property, but it doesn't use that MSBuild property....

Comment (livarcocc) [inline src/Microsoft.DotNet.Cli.Utils/CommandResolution/IProject.cs]: Sure, will do.

**Comment by livarcocc** [comment]
@piotrpMSFT Thanks Peter. I will merge this as soon as it is green to unblock the ASP.NET folks.

**Comment by livarcocc** [comment]
@dotnet-bot Test OSX x64 Release Build

**Pull request merged**